### PR TITLE
Fixes #25683 - Do not run postun script if upgrading

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: templates/postun.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+if [ $1 -eq 0 ]; then
```
Comment (ehelms) [inline templates/postun.erb]: RPM post uninstall passes arguments?

File: manifests/katello.pp [hunk 2]
```diff
@@ -55,7 +55,7 @@
     # (especially in the custom certs scenario)
     files            => ["${katello_rhsm_setup_script_location}:755=${katello_www_pub_dir}/${katello_rhsm_setup_script}"],
     bootstrap_script => "/bin/bash ${katello_rhsm_setup_script_location}",
-    postun_script    => 'test -f /etc/rhsm/rhsm.conf.kat-backup && command cp /etc/rhsm/rhsm.conf.kat-backup /etc/rhsm/rhsm.conf',
+    postun_script    => template('certs/postun.erb'),
```
Comment (ekohl) [inline manifests/katello.pp]: If there's no template I'd prefer this to be a static file. You'll need to move it to `files/postun.sh` as well.
```suggestion
    postun_script    => file('certs/postun.sh'),
```

File: templates/postun.erb (hunk 1 above)
Comment (ekohl) [reply templates/postun.erb]: Yes. https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#_syntax

Comment (jlsherrill) [reply templates/postun.erb]: was news to me too :)

File: manifests/katello.pp (hunk 2 above)
Comment (jlsherrill) [reply manifests/katello.pp]: Learned something new!  Done!

**Comment by ekohl** [comment]
I believe this fixes https://projects.theforeman.org/issues/25683. Should be move it to the installer project and properly use `Fixes #25683` in the commit message here?

**Comment by jlsherrill** [comment]
updated issue association

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**